### ISSUE TITLE
ESP32C3 Hardware SPI init error

### ISSUE THREAD

**Issue by Walkline80** [micropython/micropython#8204]
In file `ports/esp32/machine_hw_spi.c`, [here](https://github.com/micropython/micropython/blob/master/ports/esp32/machine_hw_spi.c#L85) and [here](https://github.com/micropython/micropython/blob/master/ports/esp32/machine_hw_spi.c#L270) defines the dma channel for esp32c3

```c
dma_chan = 1;
```

However, according to [this doc](https://docs.espressif.com/projects/esp-idf/en/latest/esp32c3/api-reference/peripherals/spi_master.html#_CPPv416spi_common_dma_t), esp32c3 dma channel only support value 0 and 3 (including all IDF versions support for esp32c3)

Here is my test code

```python
from machine import SPI, Pin

spi = SPI(
    1,
    baudrate=40000000,
    polarity=1,
    sck=Pin(6),
    mosi=Pin(7)
)
```

This code will cause the following error:

```docs
spi: spi_bus_initialize(634): invalid dma channel, chip only support spi dma channel auto-alloc
```

My solution is modify the file `ports/esp32/machine_hw_spi.c` [line 271](https://github.com/micropython/micropython/blob/master/ports/esp32/machine_hw_spi.c#L271) to:

```c
        #if CONFIG_IDF_TARGET_ESP32S2 || CONFIG_IDF_TARGET_ESP32S3 || CONFIG_IDF_TARGET_ESP32C3
```

**Comment by Eivs** [comment]
I have the same problem

`MicroPython v1.18 on 2022-01-17; ESP32C3 module with ESP32C3
Type "help()" for more information.
>>> from machine import Pin, SPI
>>> hw_spi = SPI(1,  sck=Pin(2),  mosi=Pin(2),  miso=Pin(10))
`

`
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: invalid configuration
`

**Comment by robert-hh** [comment]
That seems to be a different error: you assigned the same pin to sck and mosi.

**Comment by Eivs** [comment]
Thanks for the reply

I still get the same result after changing to the correct pin

<img width="1337" alt="image" src="https://user-images.githubusercontent.com/358178/156877719-7bed5471-8760-4830-9c1f-bf5d7d859b5f.png">

```
>>> hw_spi = SPI(1,  sck=Pin(2),  mosi=Pin(3),  miso=Pin(10))
  Traceback (most recent call last):
    File "<stdin>", line 1, in <module>
  OSError: invalid configuration
```

**Comment by laxtiz** [comment]
i have some board from luatos's shop, and flash that firmware [esp32c3](https://micropython.org/download/esp32c3/).

![QQ截图20220306190721](https://user-images.githubusercontent.com/3883767/156920441-f023e72e-f6d2-41e3-a2e8-1de9fc07d588.png)

**Comment by Eivs** [comment]
> i have some board from luatos's shop, and flash that firmware [esp32c3](https://micropython.org/download/esp32c3/).
> 
> ![QQ截图20220306190721](https://user-images.githubusercontent.com/3883767/156920441-f023e72e-f6d2-41e3-a2e8-1de9fc07d588.png)

HI！are you in WH?  I'm in Optics Valley .

**Comment by Walkline80** [comment]
@Eivs @laxtiz maybe you could setup debug level to `LOG_WARN` for more details ([esp.osdebug](https://docs.micropython.org/en/latest/library/esp.html?highlight=esp%20osdebug#esp.osdebug))

**Comment by Eivs** [comment]
Thanks, I got the same error

E (375164) spi: spi_bus_initialize(762): invalid dma channel, chip only support spi dma channel auto-alloc

**Comment by robert-hh** [comment]
I can make a PR using the code suggested by @Walkline80, unless it's done by her/him.

**Comment by Eivs** [comment]
On my board, test is passed，But I don't know why it only works at 40MHZ

**Comment by robert-hh** [comment]
40MHz COU freq or 40 MHz SPI clk?

**Comment by robert-hh** [comment]
I just repeated my test with a Node-MCU esp32-c3-32S-kit. SPI works fine at a CPU freq of 160 MHz and given baud rate. Test lines:
```
from machine import SPI, Pin
spi=SPI(1, sck=Pin(8), mosi=Pin(2), miso=Pin(3), baudrate=1_000_000)
spi.write(b'hello world')
b=bytearray(11)
spi.write_readinto(b'hello world', b)
```
Same with 10 MHz or 100kHz

**Comment by dpgeorge** [comment]
Fixed by 09b55dc297da2b4dc3915f2a4cc0e3b53ea07359
